lumicks/pylake/nb_widgets/fd_selector.py

```diff
@@ -0,0 +1,136 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import time
+
+
+class FdRangeSelectorWidget:
+    def __init__(self, fd_curve, ax=None, show=True):
+        self._ax = ax if ax else plt.axes(label=f"lk_widget_{time.time()}")
+        self.fd_curve = fd_curve
+        self.time = self.fd_curve.f.timestamps
+        self.connection_id = None
+        self.current_range = []
+        self.ranges = []
+        self.open = show
+
+        if show:
+            self._ax.get_figure().canvas.mpl_connect('close_event', self.close)
+            self.update_plot()
+
+    def close(self, event):
+        self.open = False
+
+    def wait_for_input(self):
+        while self.open:
+            plt.pause(.0001)
+
+    def to_seconds(self, timestamp):
+        return (timestamp - self.time[0]) / 1e9
+
+    def _add_point(self, fd_timestamp):
+        self.current_range.append(fd_timestamp)
+
+        # We have a range selected. Append it to the list.
+        if len(self.current_range) == 2:
+            self.ranges.append(np.sort(self.current_range))
+            self.current_range = []
+            self.update_plot()
+
+        # Draw a vertical line for some immediate visual feedback
+        plt.axvline(self.to_seconds(fd_timestamp))
+        self._ax.get_figure().canvas.draw()
+
+    def _remove_point(self, fd_timestamp):
+        in_range = [start < fd_timestamp < stop for start, stop in self.ranges]
+        selected = np.nonzero(in_range)[0]
+        if len(selected) > 0:
+            selected_ranges = [self.ranges[x] for x in selected]
+            smallest_segment = np.argmin([stop - start for start, stop in selected_ranges])
+            self.ranges.pop(selected[smallest_segment])
+        self.update_plot()
```

Comment:
At first glance, I don't understand this function.  
It seems to remove the smallest range that intersects with the point we pass in, but the name implies removing a point not a range, so I'm not sure what is intended.

I think either a comment or maybe renaming the function would help.

Reply:
Thanks. Yes, I agree the function name was a bit of a red :fish:. Renamed to `remove_range` to avoid the confusion. Also added a little docstring: ```"""Removes a range if the provided timestamp falls inside it."""```